tests/heap/main.c

```diff
@@ -30,12 +32,12 @@ main(int argc, char * argv[])
 		exit(1);
 
 	/* Suck in the input. */
-	while ((s = fgetln(stdin, &l)) != NULL) {
```

Comment:
updated PR; excluded the portability fixes for later.

At the moment, getting the kivaloo source and running "make" (or "bmake") on a strictly C99-compliant platforms fails.  `fgetln` and `fcloseall` are non-portable extensions.  I personally think that portability is a good thing, but I can't make a business case for it.

An alternative would be to look into supplying our own fgetln function, either copied from bsd or some other source.  For example, http://ftp.netbsd.org/pub/pkgsrc/current/pkgsrc/net/tnftp/files/libnetbsd/fgetln.c

My first thought is that `fgets` would be easier than messing around with detecting fgetln in the build system, including a replacement function, etc.  Or I could look into other C line-reading functions?  I mean, this is currently just reading a 57-line text file, with a maximum line length of 54 characters (there's one line with a tab that's visually at 57 chars, but the actual length is 50).